src/sharepoint/regionalsettings.ts

```diff
@@ -71,28 +71,87 @@ export interface RegionalSettingsProps {
 }
 
 /**
- * Describes TimeZone ODada object
+ * Describes installed languages ODada queriable collection
  */
-export class TimeZone extends SharePointQueryableInstance {
-    constructor(baseUrl: string | SharePointQueryable, path = "timezone") {
+export class InstalledLanguages extends SharePointQueryableInstance {
+    constructor(baseUrl: string | SharePointQueryable, path = "installedlanguages") {
         super(baseUrl, path);
     }
 }
 
 /**
- * Describes installed languages ODada queriable collection
+ * Describes TimeZone ODada object
  */
-export class InstalledLanguages extends SharePointQueryableInstance {
-    constructor(baseUrl: string | SharePointQueryable, path = "installedlanguages") {
+export class TimeZone extends SharePointQueryableInstance {
+    constructor(baseUrl: string | SharePointQueryable, path = "timezone") {
         super(baseUrl, path);
     }
+
+    /**
+     * Gets an Local Time by UTC Time
+     *
+     * @param utcTime UTC Time as Date or ISO String
+     */
+    public utcToLocalTime(utcTime: string | Date): Promise<{ localTime: string }> {
+        let dateIsoString: string;
+        if (typeof utcTime === "string") {
+            dateIsoString = utcTime;
+        } else {
+            dateIsoString = utcTime.toISOString();
+        }
+        return this.clone(TimeZone, `utctolocaltime(@date)?@date='${dateIsoString}'`)
+            .postCore()
+            .then(res => {
+                return {
+                    localTime: res.UTCToLocalTime,
+                };
+            });
+    }
+
+    /**
+     * Gets an UTC Time by Local Time
+     *
+     * @param localTime Local Time as Date or ISO String
+     */
+    public localTimeToUTC(localTime: string | Date): Promise<{ utcTime: string }> {
+        let dateIsoString: string;
+        if (typeof localTime === "string") {
+            dateIsoString = localTime;
+        } else {
+            dateIsoString = localTime.toISOString();
+        }
+        return this.clone(TimeZone, `localtimetoutc(@date)?@date='${dateIsoString}'`)
+            .postCore()
+            .then(res => {
+                return {
+                    utcTime: res.LocalTimeToUTC,
+                };
+            });
+    }
 }
 
 /**
  * Describes time zones queriable collection
  */
 export class TimeZones extends SharePointQueryableCollection {
-  constructor(baseUrl: string | SharePointQueryable, path = "timezones") {
-      super(baseUrl, path);
-  }
+    constructor(baseUrl: string | SharePointQueryable, path = "timezones") {
+        super(baseUrl, path);
+    }
+
+    // https://msdn.microsoft.com/en-us/library/office/jj247008.aspx - timezones ids
+    /**
+     * Gets an TimeZone by id
+     *
+     * @param id The integer id of the timezone to retrieve
+     */
+    public getById(id: number): TimeZone {

```

Comment:
Hey @patrick-rodgers, 

There is one controversial thing which I'm not completely sure in.

The endpoint of getting a timezone by id (`/regionalsettings/timezones(15)`) allows only Post method. There are a lot of `.postCore` or `.postAsCore` usages but it's for closing up statements.
But the main goal of getting by timezone Id is to chain it further, e.g. with `utcToLocalTime`, at the same time, allowing using `.get()` if we really need getting the particular time zone.

I achieved it by redefining timezones' `.get` with `.postCore` in case if timezone was gotten by id.

Could you please take a look, if it's a reliable approach?

Reply:
The way this is generally handled is that there would be a timezones collection and it would have a method getById that is a post or something similar. I don't like to redefine the get as post just because I think that makes this different from the rest of the API and it gets hard to track the cases. I'll pull down this code later today and have a look. Thanks for tackling this.